common/hyperscaler/rules/parsing_results.go

```diff
@@ -0,0 +1,112 @@
+package rules
+
+import "fmt"
+
+type ParsingResults struct {
+	resolvedRules map[string]*ParsingResult
+	uniquenessSet map[string]*ParsingResult
+	uniqueResults []*ParsingResult
+
+	Results []*ParsingResult
+}
+
+func (p *ParsingResults) Apply(entry string, rule *Rule, err error) {
+	result := NewParsingResult(entry, rule)
+
+	if err != nil {
+		result.AddParsingError(err)
+	} else {
+		if rule.IsResolved() {
+			p.resolvedRules[rule.StringNoLabels()] = result
+		}
+	}
+
+	p.Results = append(p.Results, result)
+}
+
+func NewParsingResults() *ParsingResults {
+	return &ParsingResults{
+		Results:       make([]*ParsingResult, 0),
+		resolvedRules: make(map[string]*ParsingResult),
+		uniquenessSet: make(map[string]*ParsingResult),
+		uniqueResults: make([]*ParsingResult, 0),
+	}
+}
+
+func (p *ParsingResults) IsResolved(resolvingKey string) bool {
+	_, resolvingRuleExists := p.resolvedRules[resolvingKey]
+	return resolvingRuleExists
+}
+
+func (p *ParsingResults) CheckUniqueness() {
+
+	for _, result := range p.Results {
+
+		if result.HasErrors() {
+			p.uniqueResults = append(p.uniqueResults, result)
```

Comment:
If result has errors we add it to the uniqueResults collection/slice?
It is counterintuitive IMHO.